samza-core/src/main/java/org/apache/samza/context/ContextImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.apache.samza.context;
+
+public class ContextImpl implements Context {
+  private final JobContext jobContext;
+  private final ContainerContext containerContext;
+  private final TaskContext taskContext;
+  private final ApplicationDefinedContainerContext applicationDefinedContainerContext;
+  private final ApplicationDefinedTaskContext applicationDefinedTaskContext;
+
+  /**
+   * @param jobContext non-null job context
+   * @param containerContext non-null framework container context
+   * @param taskContext non-null framework task context
+   * @param applicationDefinedContainerContext nullable application-defined container context
+   * @param applicationDefinedTaskContext nullable application-defined task context
+   */
+  public ContextImpl(JobContext jobContext, ContainerContext containerContext, TaskContext taskContext,
+      ApplicationDefinedContainerContext applicationDefinedContainerContext,
+      ApplicationDefinedTaskContext applicationDefinedTaskContext) {
+    this.jobContext = jobContext;
```

Comment:
nit: since the javadoc states "non-null", can we also add Preconditions.checkNotNull() here?